lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -307,24 +330,246 @@ class FwProgCmdProvider : public CmdProvider {
                 m_nextCmd = FLSPRG_COMPLETE;
                 while (gapFill > 0U)
                 {
-                    m_updatedCmdBuffer[WRITE_BLOCK_CMD_LENGTH + m_firmwareBytesRead] = 0xFF;
+                    m_updatedCmdBuffer[WRITE_BLOCK_CMD_LENGTH +m_firmwareBytesRead] = 0xFF;   //m_firmwareBuffer[WRITE_BLOCK_CMD_LENGTH + m_firmwareBytesRead] = 0xFF;
                     --gapFill;
                     ++m_firmwareBytesRead;
                 }
             }
             else {
                 m_nextCmd = FLSPRG_SET_ADDR;
             }
-            
             fwProgCmd.command = m_updatedCmdBuffer;
             fwProgCmd.commandsize = sizeof(m_updatedCmdBuffer);
+ 
+            m_count++;
+            LOG_DEBUG("Write count = %d", m_count);
             return true;
         }
 
         return true;
     }
 };
 
+/**
+ * @class ByteCheckCmdProvider
+ * @brief Implementation of a command provider for reading and verifying bytes.
+ *
+ * This class provides functionality to read bytes and compare them against expected bytes.
+ * It is designed to be used in a bootloader communication system to ensure the integrity
+ * of data read from a target device.
+ */
+class ByteCheckCmdProvider : public CmdProvider {
+public:
+    /**
+     * @brief Constructs a new ByteCheckCmdProvider object.
+     *
+     * @param [in] fileName The name of the file containing the expected bytes.
+     * @remarks The file specified by fileName should contain the expected bytes to be compared with.
+     */
+    ByteCheckCmdProvider(const char* fileName) :
+        m_updatedCmd(),
+        m_firmwareBuffer(),
+        m_updatedCmdBuffer{},
+        m_fileName(fileName),
+        m_firmwareFile(),
+        m_count(0U),
+        m_firmwareSize(0U),
+        m_max_buffers(0),

```

Comment:
name convention would be m_maxBuffers.  But its better called m_maxBlocks.

Reply:
Okay!